Frameworks/Foundation/NSURLRequest.mm

```diff
@@ -113,7 +113,7 @@ - (instancetype)mutableCopyWithZone:(NSZone*)zone {
 */
 - (instancetype)copyWithZone:(NSZone*)zone {
     if ([self isKindOfClass:[NSMutableURLRequest class]]) {
-        return [[NSURLRequest alloc] _initWithURLRequest:self];
+        return [[NSMutableURLRequest alloc] _initWithURLRequest:self];
```

Comment:
This is a somewhat worrying encapsulation break.
We should just override `copyWithZone` in NSMutableURLRequest.